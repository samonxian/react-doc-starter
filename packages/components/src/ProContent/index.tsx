import React from 'react';
import { Empty, Alert } from 'antd';

export interface ProContentProps {
  /**
   * 是否展示搜索提示语
   */
  showSearchTip?: boolean;
  /**
   * 搜索提示语
   */
  searchTip?: string;
  /**
   * 是否展示空数据提示语
   */
  showNoDataTip?: boolean;
  /**
   * 空数据提示语
   */
  noDataTip?: string;
  /**
   * 是否展示警告提示语
   */
  showWarnTip?: boolean;
  /**
   * 警告提示语
   */
  warnTip?: string;
  /**
   * 是否展示错误提示语
   */
  showErrorTip?: boolean;
  /**
   * 错误提示语
   */
  errorTip?: string;
  /**
   * 是否展示成功提示语
   */
  showSuccessTip?: boolean;
  /**
   * 成功提示语
   */
  successTip?: string;
  children: React.ReactNode;
}
/**
 * 内容组件，包含搜索提示、空数据提示、错误提示，如果这些其中一个设置了，子组件内容不会被渲染
 * 提示优先级
 * 错误提示 > 警告提示 > 成功提示 > 无数据提示 > 查询提示
 */
const ProContent: {
  (props: ProContentProps): JSX.Element | null;
  displayName: string;
  defaultProps?: Record<string, any>;
} = (props: ProContentProps) => {
  const {
    showSearchTip,
    searchTip,
    showNoDataTip,
    noDataTip,
    showWarnTip,
    warnTip,
    showErrorTip,
    errorTip,
    showSuccessTip,
    successTip,
  } = props;

  if (showErrorTip) {
    return <Alert type="error" message={errorTip} showIcon />;
  }

  if (showWarnTip) {
    return <Alert type="warning" message={warnTip} showIcon />;
  }

  if (showSuccessTip) {
    return <Alert type="success" message={successTip} showIcon />;
  }

  if (showNoDataTip) {
    // return <Alert type="info" message={noDataTip} showIcon />;
    return <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} description={noDataTip} />;
  }

  if (showSearchTip) {
    return <Alert type="info" message={searchTip} showIcon />;
  }

  return <>{props.children}</>;
};

ProContent.displayName = 'ProContent';

ProContent.defaultProps = {
  searchTip: '请在上方输入然后点击查询',
  noDataTip: '暂无数据',
  successTip: '成功提示',
  warnTip: '警告提示',
  errorTip: '错误提示',
};

export default ProContent;
