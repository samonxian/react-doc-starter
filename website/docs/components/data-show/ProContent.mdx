## 使用

### 基本使用

<CodeShow fileList={['$demo/ProContent/Basic.tsx', '$demo/ProContent/Basic.css']} />

## Props

<PropsTable src="$components/ProContent" showDescriptionOnSummary />
