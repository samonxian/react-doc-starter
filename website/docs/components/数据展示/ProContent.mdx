## 使用

### 基本使用

<CodeShow fileList={['$demo/ProContent/Basic.tsx']} />

## Props

<PropsTable src="$components/ProContent" showDescriptionOnSummary />
