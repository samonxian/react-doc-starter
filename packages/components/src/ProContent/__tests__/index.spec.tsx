import { expect, it } from 'vitest';
import React from 'react';
import renderer from 'react-test-renderer';
import ProContent from '../index';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
}

it('ProContent rendered', () => {
  const component = renderer.create(
    <>
      <ProContent>内容区域</ProContent>
      <ProContent showSearchTip>内容区域</ProContent>
      <ProContent showNoDataTip>内容区域</ProContent>
      <ProContent showSuccessTip>内容区域</ProContent>
      <ProContent showWarnTip>内容区域</ProContent>
      <ProContent showErrorTip>内容区域</ProContent>
    </>,
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
