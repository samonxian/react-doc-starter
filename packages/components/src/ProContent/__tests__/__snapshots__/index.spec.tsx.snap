// Vitest Snapshot v1

exports[`ProContent rendered 1`] = `
[
  "内容区域",
  <div
    className="ant-alert ant-alert-info"
    data-show={true}
    role="alert"
    style={{}}
  >
    <span
      aria-label="info-circle"
      className="anticon anticon-info-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      className="ant-alert-content"
    >
      <div
        className="ant-alert-message"
      >
        请在上方输入然后点击查询
      </div>
    </div>
  </div>,
  <div
    className="ant-empty ant-empty-normal"
  >
    <div
      className="ant-empty-image"
    >
      <svg
        className="ant-empty-img-simple"
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            className="ant-empty-img-simple-ellipse"
            cx="32"
            cy="33"
            rx="32"
            ry="7"
          />
          <g
            className="ant-empty-img-simple-g"
            fillRule="nonzero"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              className="ant-empty-img-simple-path"
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      className="ant-empty-description"
    >
      暂无数据
    </div>
  </div>,
  <div
    className="ant-alert ant-alert-success"
    data-show={true}
    role="alert"
    style={{}}
  >
    <span
      aria-label="check-circle"
      className="anticon anticon-check-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </span>
    <div
      className="ant-alert-content"
    >
      <div
        className="ant-alert-message"
      >
        成功提示
      </div>
    </div>
  </div>,
  <div
    className="ant-alert ant-alert-warning"
    data-show={true}
    role="alert"
    style={{}}
  >
    <span
      aria-label="exclamation-circle"
      className="anticon anticon-exclamation-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="exclamation-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      className="ant-alert-content"
    >
      <div
        className="ant-alert-message"
      >
        警告提示
      </div>
    </div>
  </div>,
  <div
    className="ant-alert ant-alert-error"
    data-show={true}
    role="alert"
    style={{}}
  >
    <span
      aria-label="close-circle"
      className="anticon anticon-close-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </span>
    <div
      className="ant-alert-content"
    >
      <div
        className="ant-alert-message"
      >
        错误提示
      </div>
    </div>
  </div>,
]
`;
